skore/src/skore/sklearn/_estimator/metrics_accessor.py

```diff
@@ -725,6 +920,13 @@ def custom_metric(
             The name of the metric. If not provided, it will be inferred from the
             metric function.
 
+        data_source : {"test", "train", "X_y"}, default="test"
+            The data source to use.
+
+            - "test" : use the test set provided when creating the reporter.
+            - "train" : use the train set provided when creating the reporter.
+            - "X_y" : use the provided `X` and `y` to compute the metric.
+
```

Comment:
Can you add documentation for `data_source_hash`?

Reply:
You mean on each private functions (because we don't expose it on the public one)?

Reply:
Since they are private and only the this particular parameter change, we can find a middle ground to not repeat everything:

```
"""Private interface of {metrics} to be able to pass `data_source_hash`.

`data_source_hash` is either an `int` when we already computed the hash
and are able to pass it around or `None` and thus trigger its computation
in the underlying process.
"""
```

Would this fit what you had in mind @augustebaum?

Reply:
Yep that works